src/main/java/biz/netcentric/maven/extension/repofromenv/FromEnvReposConfigurationProcessor.java

```diff
@@ -179,9 +190,43 @@ private Server getServer(RepoFromEnv repoFromEnv) {
         server.setId(repoFromEnv.getId());
         server.setUsername(repoFromEnv.getUsername());
         server.setPassword(repoFromEnv.getPassword());
+        if(repoFromEnv.isUsePreemptiveAuth()) {
+            // configure preemptive auth for wagon transport (< Maven 3.9.0)
+            configurePreemptiveAuthForWagon(server);
+        }
         return server;
     }
 
+    void configurePreemptiveAuthForMavenResolver(CliRequest cliRequest, List<RepoFromEnv> reposFromEnv) {
+        
+        reposFromEnv.stream()
```

Comment:
You should use the same approach for configuring Wagon/Resolver. Look at 
https://github.com/apache/maven/blob/master/impl/maven-core/src/main/java/org/apache/maven/internal/aether/DefaultRepositorySystemSessionFactory.java how Maven specific settings are converted to Resolver specific properties.